### PR TITLE
[Tests] add tests for IPC

### PR DESCRIPTION
**Pull request by floriantschopp**
# Description

Adds a unit test for pub/sub and service/client usage.

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: modules/examples/tests/zenoh_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  auto subscriber = ipc::subscribe<heph::ipc::zenoh::Subscriber, Pose>(
+      session, topic,
+      [&received_message]([[maybe_unused]] const ipc::MessageMetadata& metadata,
+                          const std::shared_ptr<Pose>& message) { received_message = *message; });
+
+  const auto success = publisher.publish(send_message);
+  EXPECT_TRUE(success);
+
+  // Give enough time for the message to be received
+  std::this_thread::sleep_for(std::chrono::seconds(1));
```
Comment (filippobrizzi) [inline modules/examples/tests/zenoh_tests.cpp:43]: I think 1ms should be enough here

File: modules/examples/tests/zenoh_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  ipc::Config server_config{};
+  auto server_session = ipc::zenoh::createSession(std::move(server_config));
+  auto service_server = ipc::zenoh::Service<Pose, Pose>(server_session, service_topic,
+                                                        [](const Pose& request) { return request; });
+
+  ipc::Config client_config{};
+  auto client_session = ipc::zenoh::createSession(std::move(client_config));
+  const auto reply = ipc::zenoh::callService<Pose, Pose>(*client_session, service_topic, request_message,
+                                                         std::chrono::milliseconds(1000));
```
Comment (filippobrizzi) [inline modules/examples/tests/zenoh_tests.cpp:65]: also here the timeout it is too long

**Review by filippobrizzi — COMMENTED** [review]
this test should go in the `ipc` package, but to do that we would need to first create the `test_helper` package to provide with dummy proto messages.
I would say, let's leave it here with a comment and let's add a task in the project to move it when we have the test helper

File: modules/examples/tests/zenoh_tests.cpp (hunk 2 above)
Comment (floriantschopp) [reply modules/examples/tests/zenoh_tests.cpp]: 1 ms was not long enough...

**Review by filippobrizzi — APPROVED** [review]

**Pull request merged**